vpr/src/route/rr_graph.cpp

```diff
@@ -3211,6 +3259,45 @@ static void build_rr_chan(RRGraphBuilder& rr_graph_builder,
     }
 }
 
+static void build_inter_die_custom_sb_rr_chan(RRGraphBuilder& rr_graph_builder,
+                                              const int layer,
+                                              const int x_coord,
+                                              const int y_coord,
+                                              const int const_index_offset,
+                                              const t_chan_width& nodes_per_chan,
+                                              const t_chan_details& chan_details_x) {
+    auto& mutable_device_ctx = g_vpr_ctx.mutable_device();
+    const t_chan_seg_details* seg_details = chan_details_x[x_coord][y_coord].data();
+
+    /* Loads up all the extra routing resource nodes required to support multi-die custom switch blocks */
```

Comment:
Expand this comment explaining what is going on.  Nodes have already been created in the lookup so you're just going through the lookup until you run out of nodes; you create all these extra nodes. The extra nodes are ...